### PR TITLE
Add scheduled and immediate notification sending

### PR DESCRIPTION
**Pull request by MJDeligan**
Changes:

- 2 new edge functions one for singular notifications that are to be sent immediately (use with db webhooks) and one for batch scheduled notifications that are sent from a cron-job
- Added an Sql-Procedure that gets and stringifies due notifications, sends them to the batch edge function then deletes them (as per @stefnotch 's idea)
- Added a cron job that calls the new sql-procedure every minute

### CODE REVIEW

**Comment by MJDeligan** [comment]
GitGuardian just detected the public key. Not an issue

File: frontend/src/stores/user.ts [hunk 1]
```diff
@@ -41,60 +35,32 @@ export const useUserStore = defineStore("user", () => {
     user.value = session?.user ?? null;
   });
 
-  /**
-   * CRUD OPERATIONS
-   */
-  const createEmptyProfile = async () => {
```
Comment (MJDeligan) [inline frontend/src/stores/user.ts:47]: The profile is created by a trigger when the user is registered

File: frontend/src/stores/user.ts [hunk 2]
```diff
@@ -8,31 +8,25 @@ import { acceptHMRUpdate, defineStore } from "pinia";
...
+export interface Profile {
   email?: string; // derived from "user"
-  username?: string;
+  fullName?: string;
   description?: string;
-  avatar_url?: string;
 }
 
 export const useUserStore = defineStore("user", () => {
-  const emptyProfile = (): Profile => {
```
Comment (MJDeligan) [inline frontend/src/stores/user.ts:19]: This basically just returns an empty object

File: backend/supabase/functions/sendBatchNotifications/index.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        end += chunkSize;
+    }
+    return ret;
+}
+
+serve(async (req: Request) => {
+    const functionSecret = Deno.env.get('FUNCTION_SECRET');
+    const requestSecret = req.headers.get('function-secret');
+
+    if (functionSecret !== requestSecret) {
```
Comment (stefnotch) [inline backend/supabase/functions/sendBatchNotifications/index.ts:28]: How secret is our secret? Depending on the answer, we might want to use https://deno.land/std@0.152.0/crypto/timing_safe_equal.ts?s=timingSafeEqual

(Which is a super recent API addition  https://github.com/denoland/deno_std/pull/2503 )

File: frontend/src/stores/user.ts [hunk 4]
```diff
@@ -41,60 +34,32 @@ export const useUserStore = defineStore("user", () => {
...
-      throw new Error("user.ts: User not loaded");
+    if (!user.value) throw new Error("user.ts: User not loaded");
     console.log("user.ts: Fetching profile with email:", user.value.email);
 
     // this will still end up logging a red line in the console
     const { data, error } = await supabase
       .from("profiles")
-      .select(`username, description, avatar_url`)
-      .eq("id", user.value.id)
+      .select(`full_name, description, email`)
```
Comment (stefnotch) [inline frontend/src/stores/user.ts:45]: Happy news here: This is going to become type-safer in the future
https://github.com/supabase/postgrest-js/pull/279

(Yes Typescript is powerful enough to check arbitrary strings. Go figure.)

**Review by stefnotch — APPROVED** [review]

**Pull request merged**